models/inventory/StockManager.ts

```diff
@@ -126,6 +127,58 @@ export class StockManager {
     throw new ValidationError(t`Both From and To Location cannot be undefined`);
   }
 
+  async #validateBatchWiseStockAvailability(details: SMIDetails) {
+    /*
+     * Checks if hasBatchNumber is enabled in the item
+     * If user has not entered batchNumber raises a ValidationError
+     * If entered quantity is greater than the available quantity in the batch, raises a ValidationError
+     */
+
+    if (!details.fromLocation) {
+      return;
+    }
+
+    const ifItemHasBatchNumber = await this.fyo.getValue(
+      'Item',
+      details.item,
+      'hasBatchNumber'
+    );
+    const date = details.date.toISOString();
+    const formattedDate = this.fyo.format(details.date, 'Datetime');
```

Comment:
Define variables closer to where they're used; `formattedDate` is not even used unless you have to throw the validation error.

Reply:
corrected